### PR TITLE
chore: Update Environment Model

### PR DESCRIPTION
**Pull request by aws-tyler**
Issue #, if available:
GALI-1789
GALI-1790

Description of changes:

* Removed datasetIds from Environment object.
* Changed type on Environment to be the sk value of the EnvironmentTypeConfig.
* Updated tests.

Checklist:

* [X] Have you successfully deployed to an AWS account with your changes?
* [X] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts [hunk 1]
```diff
@@ -129,9 +128,8 @@ describe('SagemakerNotebookEnvironmentLifecycleService', () => {
     expect(response).toEqual({ ...environment, status: 'PENDING' });
   });
 
-  test('Launch should return mocked id when mounting datasets', async () => {
+  test.skip('Launch should return mocked id when mounting datasets', async () => {
```
Comment (aws-tyler) [inline solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts:131]: @nguyen102 I'm pretty sure this test (and possibly the one above it) isn't doing what we think it is, so I marked it as `skip` but I could also remove it completely. We are mocking `environment` which gets passed into `sm.launch`, is never modified, gets sent back in the response, and then is checked against its original value. Is that accurate or am I maybe missing something?

File: solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts [hunk 2]
```diff
@@ -129,9 +128,13 @@ describe('SagemakerNotebookEnvironmentLifecycleService', () => {
...
     const envHelper = new EnvironmentLifecycleHelper();
-    environment.datasetIds = ['exampleDS'];
+    environment.DATASETS = [
+      {
+        id: 'dataset-123'
```
Comment (aws-tyler) [inline solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts:135]: only included the `id` here and not the whole object since the rest isn't needed for this test.

File: workbench-core/environments/src/services/environmentService.test.ts [hunk 3]
```diff
@@ -1038,7 +1040,8 @@ describe('EnvironmentService', () => {
         PROJ: projItem,
         ...env,
         provisionedProductId: '',
-        error: undefined
+        error: undefined,
+        datasetIds: undefined
```
Comment (maghirardelli) [inline workbench-core/environments/src/services/environmentService.test.ts:1044]: Why do we need `datasetIds: undefined`?

File: workbench-core/environments/src/services/environmentService.ts [hunk 4]
```diff
@@ -396,15 +394,14 @@ export class EnvironmentService {
...
-      datasetIds: params.datasetIds,
       envTypeConfigId: params.envTypeConfigId,
       updatedAt: new Date().toISOString(),
       updatedBy: user.id,
       createdAt: new Date().toISOString(),
       createdBy: user.id,
       owner: user.id,
       status: params.status || 'PENDING',
-      type: params.envTypeId,
+      type: '',
```
Comment (maghirardelli) [inline workbench-core/environments/src/services/environmentService.ts:404]: Why don't we set this to be `${resourceTypeToKey.envType}#${params.envTypeId}${resourceTypeToKey.envTypeConfig}#${params.envTypeConfigId}` like in line 375?

File: workbench-core/environments/src/utilities/environmentLifecycleHelper.ts [hunk 5]
```diff
@@ -142,13 +142,13 @@ export default class EnvironmentLifecycleHelper {
 
   /**
    * Delete the access points for all datasets attached to a given workspace.
-   * @param envMetadata - the environment for which access point(s) were created
+   * @param environment - the environment for which access point(s) were created
```
Comment (maghirardelli) [inline workbench-core/environments/src/utilities/environmentLifecycleHelper.ts:145]: environment.DATASETS does not need to be defined (see Environment type). Does this handle that case?

**Review by FernandoAranda — APPROVED** [review]

File: solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.ts [hunk 6]
```diff
@@ -32,8 +32,13 @@ export default class SagemakerNotebookEnvironmentLifecycleService implements Env
 
     const { datasetsBucketArn, mainAccountRegion, mainAccountId, mainAcctEncryptionArn } =
       await this.helper.getCfnOutputs();
+
+    const datasetIds: string[] =
+      envMetadata.DATASETS?.map((dataSet: { id: string }) => {
```
Comment (maghirardelli) [inline solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.ts:37]: Did you add a test for multiple datasets as well?

File: solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts (hunk 1 above)
Comment (maghirardelli) [reply solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts:131]: Let's not get rid of test cases. I think this and the test above it are important. Can you fix this test to do what you think it needs to? Why does this not do what it should?

File: solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts [hunk 7]
```diff
@@ -149,6 +152,8 @@ describe('SagemakerNotebookEnvironmentLifecycleService', () => {
     const sm = new SagemakerNotebookEnvironmentLifecycleService();
     sm.helper = envHelper;
     const response = await sm.launch(environment);
+
+    // This does not appear to be testing what the name claims it is and should be reviewed
```
Comment (pbabz) [inline solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts:156]: maybe preface the comment with `TODO:` ... I'm not sure how we keep track of things like this :/

File: solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts (hunk 1 above)
Comment (pbabz) [reply solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts:131]: Not a fan of skipped tests ... Can we add a story to the backlog to come back and polish it if it's not testing the right thing?

File: workbench-core/environments/src/utilities/environmentLifecycleHelper.ts (hunk 5 above)
Comment (maghirardelli) [reply workbench-core/environments/src/utilities/environmentLifecycleHelper.ts:145]: For consistency if you are going to rename this param, please rename the other params in this file that are used in the same way.

**Review by pbabz — APPROVED** [review]
Looks good!

File: workbench-core/environments/src/services/environmentService.test.ts (hunk 3 above)
Comment (aws-tyler) [reply workbench-core/environments/src/services/environmentService.test.ts:1044]: Its a check to verify `datasetIds` does not appear in the final object (part of the AC)

File: solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts (hunk 7 above)
Comment (aws-tyler) [reply solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts:156]: This is temporary and won't be getting merged in, just meant to help jump start the conversation about why this test got marked `skip` and what to do about it.

File: workbench-core/environments/src/utilities/environmentLifecycleHelper.ts (hunk 5 above)
Comment (aws-tyler) [reply workbench-core/environments/src/utilities/environmentLifecycleHelper.ts:145]: Lodash returns `true` for `_.isEmpty(null)` and `_.isEmpty(undefined)` so we'll still get the desired behavior.

File: workbench-core/environments/src/services/environmentService.test.ts (hunk 3 above)
Comment (maghirardelli) [reply workbench-core/environments/src/services/environmentService.test.ts:1044]: Would not having the line also verify that? If there is a `datasetIds` value, the `expect` would fail, right?

Comment (aws-tyler) [reply workbench-core/environments/src/services/environmentService.test.ts:1044]: `toEqual` only checks that there is field equality for the fields that exist. Not including a value for `datasetIds` would mean it wouldn't get checked. By defining it as `undefined`, we're explicitly saying that we want to make sure that `environment.datasetIds` is `undefined` on both objects.

File: solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.ts (hunk 6 above)
Comment (aws-tyler) [reply solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.ts:37]: I think we only need to worry about the case where its empty (which is tested in `Launch should return a mocked id` and where its non-empty. If we've got a reason to test specific sizes though just let me know and I'm happy to add it.

File: solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts (hunk 1 above)
Comment (aws-tyler) [reply solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts:131]: Spoke with @SanketD92. My code changes actually make this test obsolete, which is why it didn't appear to be doing anything. Removed the test now that it isn't needed anymore.

Comment (SanketD92) [reply solutions/swb-reference/src/environment/sagemakerNotebook/sagemakerNotebookEnvironmentLifecycleService.test.ts]: Looks like we're mocking the `getDatasetsToMount` helper method in this test case and the one above to return `s3Mounts` and `iamPolicyDocument` values, when only one test case (this one) should have returned them (while the other returned empty strings).

**Review by pbabz — APPROVED** [review]

**Review by nguyen102 — APPROVED** [review]
